### PR TITLE
feat(access): Add constraint to check if user ID is in a gsheet list

### PR DESCRIPTION
**Pull request by patrickvenetz**
Constraint checks if granter's user ID is to be found in a `gsheets` record. If not found, constraint will fail. Contraint will hinder display of campaign.

This helps to render campaigns only to users listed in a spreadsheet.

### CODE REVIEW

File: packages/access/lib/constraints/granterInGsheet.js [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+  const { pgdb } = context
+
+  const {
+    name, // gsheets.name field
+    identifier, // row which contains a user ID
+    flag // (optional) row which all enables or disables campaign ("TRUE", "FALSE")
+  } = settings
+
+  if (!name) {
+    debug('settings.name property is falsy')
```
Comment (patte) [inline packages/access/lib/constraints/granterInGsheet.js:29]: why not throw or `return false` here?

File: packages/access/lib/constraints/granterInGsheet.js [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+  }
+
+  if (!identifier) {
+    debug('settings.identifier property is falsy')
+  }
+
+  const users = await pgdb.public.gsheets.findOneFieldOnly({ name }, 'data')
+
+  if (!users) {
+    debug(`Specified gsheet "${name}" is either missing or emmpty`)
```
Comment (patte) [inline packages/access/lib/constraints/granterInGsheet.js:39]:
```suggestion
    debug(`Specified gsheet "${name}" is either missing or empty`)
```

File: packages/access/lib/constraints/granterInGsheet.js [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+ * }
+ */
+
+const isGrantable = async (args, context) => {
+  const { granter, settings } = args
+  const { pgdb } = context
+
+  const {
+    name, // gsheets.name field
+    identifier, // row which contains a user ID
```
Comment (patte) [inline packages/access/lib/constraints/granterInGsheet.js:24]:
```suggestion
    identifier, // column which contains a user ID
```

File: packages/access/lib/constraints/granterInGsheet.js [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+    flag // (optional) row which all enables or disables campaign ("TRUE", "FALSE")
```
Comment (patte) [inline packages/access/lib/constraints/granterInGsheet.js:25]:
```suggestion
    flag // (optional) column which overwrites grant for user ("TRUE", "FALSE")
```

File: packages/access/lib/constraints/granterInGsheet.js (hunk 3 above)
Comment (patte) [reply packages/access/lib/constraints/granterInGsheet.js:24]: `idColumn` or `idColumnName` would have been a better name IMHO.

File: packages/access/lib/constraints/granterInGsheet.js [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+ *  }
+ * }
+ */
+
+const isGrantable = async (args, context) => {
+  const { granter, settings } = args
+  const { pgdb } = context
+
+  const {
+    name, // gsheets.name field
```
Comment (patte) [inline packages/access/lib/constraints/granterInGsheet.js:23]: `sheetName` would have been a better name IMHO.

File: packages/access/lib/constraints/granterInGsheet.js (hunk 4 above)
Comment (patte) [reply packages/access/lib/constraints/granterInGsheet.js:25]: `flagColumnName` i'd liked more.

**Comment by patrickvenetz** [comment]
Project aborted most successfully and adding this feature would be unnecessary dead code, hence: Closing PR.

**Pull request closed without merging**